core/ui_mixins/quick_panel.py

```diff
@@ -502,15 +502,22 @@ def on_selection_async(self, commit):
         self.on_done(commit)
 
 
-def show_stash_panel(git, on_done):
+def show_stash_panel(on_done, **kwargs):
     """
     Display stash entries in quick panel with pagination, and execute on_done(stash)
     when item is selected. `entries` can be either a list or a generator of LogEnty.
     """
 
-    lp = PaginatedPanel(
-        git.get_stashes(),
-        on_done,
-        format_item=(lambda entry: entry.id + " " + entry.description))
-    lp.show()
-    return lp
+    sp = StashPanel(on_done, **kwargs)
+    sp.show()
+    return sp
+
+
+class StashPanel(PaginatedPanel, GitCommand):
```

Comment:
I thought you wanted to avoid creating this class

Reply:
Oh, I meant either creating  `StashPanel` which computes stashes directly, or just use `show_paginated_panel`. But you were in between, your `StashPanel` still needs the list of stashes as an argument.